src/Backups/BackupEntriesCollector.cpp

```diff
@@ -251,12 +250,72 @@ bool BackupEntriesCollector::tryGatherMetadataAndCompareWithPrevious(std::option
             throw;
 
         inconsistency_error = e;
-        return false;
+        need_another_attempt = true;
```

Comment:
If `compare_collected_metadata` is not true, then we don't need another attempt right ?

Reply:
The functions `gatherTablesMetadata()` and `gatherDatabasesMetadata()` can throw the `INCONSISTENT_METADATA_FOR_BACKUP`, see for example https://github.com/ClickHouse/ClickHouse/blob/af8b1e4978519b23a8c550bd3df25a354ed799ca/src/Databases/DatabasesCommon.cpp#L352. In that case the gathered data are not consistent by itself and we need another attempt even if `compare_collected_metadata` is true.

Reply:
I've added this comment to the code.